exercises/slot_machine.livemd

```diff
@@ -0,0 +1,77 @@
+# Slot Machine
+
+## Navigation
+
+[Return Home](../start.livemd)<span style="padding: 0 30px"></span>
+[Report An Issue](https://github.com/DockYard-Academy/beta_curriculum/issues/new?assignees=&labels=&template=issue.md&title=)
+
+## The House Inc
+
+You have been hired at The House Inc.
+
+The House Inc is a casino right in the middle of Generic City. You aren't sure why a casino needs a programmer but you were desperate and applied anyway.
+
+Somehow, you actually made it past the other applicants and got the job... Or maybe that was because you were the only applicant.
+
+Your first assignment was to write the code for a slot machine. You looked at the specification and you were alarmed at how unfair this slot machine is. The House Inc isn't known for being a good place to work but you decide to talk to your boss about it anyway.
+
+So you went to your boss's office to voice your opinions on the topic and after you give a passionate speech on the value of ethics and morals. Your boss interrupts you in the middle of your ~~tantrum~~ speech and asks what ethics means.
+
+At that very moment you knew that there was no way these guys would listen because the only ethics they believe in are profit maximization.
+
+## Slot Machine Specs
+
+The slot machine is a `GenServer` that has the state of `balance` within it. Balance is how much money the machine has.
```

Comment:
Small tone suggestion, I usually start the exercise by telling the student what they are going to do. Then it's more clear if the exercise has parts that are explanations, and parts that are instructions to perform an action.

I.e. You're going to build a `SlotMachine` `GenServer` that represents a slot machine game.
The `SlotMachine` will store an initial balance of `7000` pennies. (consider changing this value)

When we play the `SlotMachine` it should have a `1` in `3` chance of paying out a reward. If it pays out a reward, the reward should be a random value between `1` and the remaining balance of the machine. The reward cannot be larger than the current balance on the machine.

```elixir
{:ok, slot_machine_pid} = SlotMachine.start_link([])

SlotMachine.play(slot_machine_pid)
# no reward, return `0`.
0

SlotMachine.play(slot_machine_pid)
# reward, randomly generated 3000 pennies
3000

# subtract the reward from the current machine balance.
4000 = :sys.get_state(slot_machine_pid)
```

Reply:
This exercise might also be clearer if instead of paying out a random reward, it always paid out the same amount. 🤔

Reply:
I think that defeats the purpose of the exercise: Mixing other concepts with GenServers. But if you believe it will add value to students if we made it a fixed reward, I'll do that.

Reply:
What about making it a bonus to the exercise?
The randomness makes it hard to document the functions.